portfolio.py

```diff
@@ -2,21 +2,49 @@
 Allocation of the asset.
 """
 
+import numpy as np
+from thousandaire.constants import TRADING_INSTRUMENTS
+
 class Portfolio(dict):
     """
     Set the portfolio
     """
-    def __init__(self, date):
+    def __init__(self, target=None, np_array=None):
         dict.__init__(self)
-        self.date = date
+        if target is not None:
+            if isinstance(np_array, np.ndarray):
+                self.decode_nparray(target, np_array)
+            else:
+                raise TypeError(
+                    'The input is %s, not a numpy array.' %(type(np_array)))
+        elif np_array is not None:
+            raise KeyError('Please input your decode target.')
+
+    def decode_nparray(self, target, np_array):
+        """
+        Decode numpy array into portfolio.
+        """
+        if len(np_array) != len(TRADING_INSTRUMENTS[target]):
+            raise ValueError(
+                'Input dimension does not match number of instruments.')
+        for index, instrument in TRADING_INSTRUMENTS[target]:
+            self[instrument] = np_array[index]
+
+    def encode_to_nparray(self, target):
+        """
+        Encode the positions into numpy type.
+        """
+        return np.array(
+            self.get(instrument, 0)
+            for instrument in TRADING_INSTRUMENTS[target])
 
-    def normalize(self):
+    def normalize(self, date):
         """
         Normalize positions to make sure the summation of absolute values is 1.
         The simulator will call this method.
         """
         position_sum = sum(map(abs, self.values()))
         if position_sum == float(0):
-            raise ValueError("zero position on %s" % (self.date))
```

Comment:
I don't think we need to pass `date` just for this error message.
We should raise errors about date information in the higher level.